PubSub/src/Topic.php

```diff
@@ -125,6 +147,22 @@ public function __construct(
         $this->info = $info;
         $this->clientConfig = $clientConfig;
 
+        if (isset($info['enableCompression']) && $info['enableCompression'] === true) {
+            $this->enableCompression = true;
+            $this->compressionBytesThreshold = self::DEFAULT_COMPRESSION_BYTES_THRESHOLD;
+            if (isset($info['compressionBytesThreshold'])) {
+                if (!is_int($info['compressionBytesThreshold'])) {
+                    throw new InvalidArgumentException(
+                        '`compressionBytesThreshold` must be an integer'
+                    );
+                } else {
+                    $this->compressionBytesThreshold = $info['compressionBytesThreshold'];
+                }
+            }
```

Comment:
Since we already documented that this is an int param, checking against non-int values is an overkill.
```suggestion
            $this->compressionBytesThreshold = $info['compressionBytesThreshold'] ?? self::DEFAULT_COMPRESSION_BYTES_THRESHOLD;
```

Reply:
PHP won't throw any error / warning if we supply a string there and eventually it would try to compare int and string which would convert one of them to the other. 

This felt little uncontrolled and thus I put that check being skeptical🫠. I do agree as that's overkill because any string would get converted to number or vice versa while comparing.

Thanks for pointing this out @vishwarajanand😄